container.go

```diff
@@ -264,3 +265,12 @@ func (c *ContainerRequest) validateMounts() error {
 	}
 	return nil
 }
+
+func RunningInContainer() bool {
+	return fileExists("/.dockerenv") || fileExists("/run/.containerenv")
```

Comment:
If I remember correctly it is not only used in testing but also to determine if the container environment information should be fetched.

I was wondering as well if it makes sense to expose this but I thought it could be useful and it's not a huge 'burden' to maintain :smile:

Regarding other indicators: I'm not sure how Docker is handling the same with Windows containers so there might be another check in the future to cover these as well (although supporting Windows containers will be fun anyway if this is something that should be done in the future) but apart from that I hope `/run/.containerenv` is not only used by Podman but by all other container providers apart from Docker

Reply:
Be careful about `.dockerenv`. To my last knowledge, this is an undocumented implementation detail of Docker and not necessarily a stable indicator. There is some discussion around this on SO, which sadly does not provide a better alternative: https://superuser.com/questions/1021834/what-are-dockerenv-and-dockerinit

Reply:
I am ware, although Podman explicitly adopted this with their `/run/.containerenv` flag.
Surely this is not the greatest solution but a fairly easy one and for instance also used in [`tc-dotnet`](https://github.com/testcontainers/testcontainers-dotnet/blob/ff76e0fb4ff00bc81b657fda982ab8c1608f6fd3/src/Testcontainers/Clients/TestcontainersClient.cs).
Furthermore the SO link you attached mentions a moby repository where they added the [exact same behavior](https://github.com/moby/libnetwork/blob/b3507428be5b458cb0e2b4086b13531fb0706e46/drivers/bridge/setup_bridgenetfiltering.go#L160) so it seems this flag is considered stable enough by the Docker guys :smile: 

An alternative approach could be to not use these flags at all but to check if `/etc/hostname` is mounted (by checking the `/proc/self/mountinfo`) and if so try to get the container ID from the mount root.

Both solutions are not perfect, I'm open for suggestions